Add masking to streamlit img2img

briansemrau opened the pull request:
There are still some things to be done before this is "user friendly"

Some settings should only be available for different modes, e.g.
- `Find Noise Steps` should only appear when Noise Mode = Find Noise
- `denoising strength` should be forced to 1.0 when using Find Noise

ZeroCool940711 commented:
Removing the form has some problems like the whole page reloading when any slider or button change, this also stops the generation, Im trying to find a fix for that but for now we do need the form, not having a form also means we have to change every variable so they are not repeated between tabs, it might take a while for those fixes which is why I havent done it, I recommend you put the form back for now.

briansemrau commented:
Didn't realize it stops generation, that's no good.

ZeroCool940711 commented:
I think I have a workaround for it but have to test it to make sure it actually works, the idea is to disable re-run on the config file so its not done on changes which would be the same as if you click on `Always Rerun` when it asks, instead of it done automatically we can do it manually when it asks at the top or even have parts of the code that can be rerun as streamlit has a function for that so if we need to redraw a part of the page we can do that, it does need to be tested, I might be wrong and it might not work.

briansemrau commented:
I re-added the form. I also added a "Refresh" button as a workaround so you can see the effect of your mask to make sure everything is in order.

Also added find_noise_for_image and another method that might be called matched noise (fourier transform method to setup the image for better inpainting)

briansemrau commented on scripts/webui_streamlit.py:
```diff
@@ -1118,21 +1124,27 @@ def process_images(
+				x = torch.cat(batch_size * [find_noise_for_image.find_noise_for_image(
+					st.session_state["model"], st.session_state["device"],
+					init_img.convert('RGB'), '', find_noise_steps, 0.0, normalize=True,
+					generation_callback=generation_callback,
```
should remove the generation callback param. it's not setup to work like this, I just liked having the preview.
you'd see "30/30" and it just keeps running when the find_noise_steps is >100

perhaps the generation callback can be reworked a tad to be more flexible

ZeroCool940711 replied:
Oh, you can do like I did on the txt2vid tab, [check this](https://github.com/sd-webui/stable-diffusion-webui/blob/ba63fd7004d397627b9bfd0ab0b4eafc65fe677d/scripts/webui_streamlit.py#L660).

ZeroCool940711 reviewed (approved):
LGTM
